tiledb/sm/rest/rest_client_remote.cc

```diff
@@ -1544,17 +1539,15 @@ const RestCapabilities& RestClientRemote::get_capabilities_from_rest() const {
 
   Buffer data;
   try {
-    throw_if_not_ok(
-        curlc.get_data(stats_, url, serialization_type_, &data, {}));
-  } catch (const std::exception& e) {
-    std::string msg = e.what();
-    if (msg.find("HTTP code 404") != std::string::npos) {
+    curlc.get_data(stats_, url, serialization_type_, &data, {});
```

Comment:
Throwing an exception might have ripple effects especially in tests (we have seen this when we un-statused VFS implementations). If this occurs, we can add an overload to `get_data` that takes an `int& status_code` parameter.